kedro-datasets/kedro_datasets/api/api_dataset.py

```diff
@@ -76,30 +110,64 @@ def __init__(
             credentials: Allows specifying secrets in credentials.yml.
                 Expected format is ``('login', 'password')`` if given as a tuple or list.
                 An ``AuthBase`` instance can be provided for more complex cases.
+            method: The Method of the request, GET, POST, PUT,
+            DELETE, HEAD, etc... data: The request payload, used for POST, PUT, etc
+            requests
+                https://requests.readthedocs.io/en/latest/user/quickstart/#more-complicated-post-requests
+            params: The url parameters of the API.
+                https://requests.readthedocs.io/en/latest/user/quickstart/#passing-parameters-in-urls
+            headers: The HTTP headers.
+                https://requests.readthedocs.io/en/latest/user/quickstart/#custom-headers
+            auth: Anything ``requests`` accepts. Normally it's either ``('login',
+            'password')``,
+                or ``AuthBase``, ``HTTPBasicAuth`` instance for more complex cases. Any
+                iterable will be cast to a tuple.
+            json: The request payload, used for POST, PUT, etc requests, passed in
+                to the json kwarg in the requests object.
+                https://requests.readthedocs.io/en/latest/user/quickstart/#more-complicated-post-requests
+            timeout: The wait time in seconds for a response, defaults to 1 minute.
+                https://requests.readthedocs.io/en/latest/user/quickstart/#timeouts
+            credentials: same as ``auth``. Allows specifying ``auth`` secrets in
+                credentials.yml.
+            save_args: Options for saving data on server. Includes all parameters used
+                during load method. Adds an optional parameter, ``chunk_size`` which determines the
+                size of the package sent at each request.
         Raises:
             ValueError: if both ``auth`` in ``load_args`` and ``credentials`` are specified.
         """
         super().__init__()
 
-        self._load_args = load_args or {}
-        self._load_args_auth = self._load_args.pop("auth", None)
+        # GET method means load
+        if method == "GET":
+            self._params = load_args or {}
 
-        if credentials is not None and self._load_args_auth is not None:
+        # PUT, POST, DELETE means save
+        elif method in ["PUT", "POST", "DELETE"]:
```

Comment:
It seems a bit odd that "DELETE" also means save. I think it makes more sense if we just don't support DELETE.

Reply:
I would also be in favor of not supporting DELETE - unless someone request it since it doesn't fit into the data catalogue I/O nicely.